pkg/aggregator/context_resolver.go

```diff
@@ -22,6 +23,7 @@ type Context struct {
 	taggerTags *tags.Entry
 	metricTags *tags.Entry
 	noIndex    bool
+	origin     string // TODO(remy): consider a string interner here
```

Comment:
I wonder what the impact of just having this field would be when the agent is handling large number of metrics?